libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -464,6 +472,30 @@ bool AP_BattMonitor::arming_checks(size_t buflen, char *buffer) const
     return true;
 }
 
+// Check's each smart battery instance for its powering off state and broadcasts notifications
+void AP_BattMonitor::checkPoweringOff(void)
+{
+    static bool powerOffNotified[AP_BATT_MONITOR_MAX_INSTANCES];
+
+    for (uint8_t i = 0; i < _num_instances; i++) {
+        if (is_powering_off(i) && !powerOffNotified[i]) {
+            // Set the AP_Notify flag, which plays the power off tones
+            AP_Notify::flags.powering_off = true;
+
+            // Send a Mavlink broadcast announcing the shutdown
+            mavlink_message_t msg;
+            mavlink_command_long_t cmd_msg;
+            cmd_msg.command = MAV_CMD_POWER_OFF_INITIATED;
+            mavlink_msg_command_long_encode(mavlink_system.sysid, MAV_COMP_ID_ALL, &msg, &cmd_msg);
+            GCS_MAVLINK::send_to_components(&msg);
```

Comment:
@peterbarker does this check for buffer space, or could this shove a partial packet into the outgoing messages?

Reply:
.... but it does drop entire packets...